puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-katello-test.yml

```diff
@@ -1,46 +1,91 @@
+---
 - job-template:
-    name: 'foreman-katello-{version}-test'
-    project-type: pipeline
+    name: foreman-katello-{version}-{distro}-{action}-test
+    project-type: 'pipeline'
     sandbox: true
+    parameters:
+      - string:
+          name: version
+          default: '{version}'
+      - string:
+          name: distro
+          default: '{distro}'
+      - string:
+          name: action
+          default: '{action}'
+      - string:
+          name: type
+          default: 'katello'
     dsl:
       !include-raw:
-        - 'pipelines/lib/foremanCentosJob.groovy{empty}'
-        - 'pipelines/lib/pipelines.groovy{empty}'
-        - 'pipelines/test/katello/{version}.groovy'
-        - '../theforeman.org/pipelines/lib/ansible.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/duffy.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}'
-
-- job-template:
-    name: 'foreman-katello-upgrade-{version}-test'
-    project-type: pipeline
-    sandbox: true
-    dsl:
-      !include-raw:
-        - 'pipelines/lib/foremanCentosJob.groovy{empty}'
-        - 'pipelines/lib/pipelines.groovy{empty}'
-        - 'pipelines/test/katello/{version}-upgrade.groovy'
-        - '../theforeman.org/pipelines/lib/ansible.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/duffy.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}'
+        - pipelines/lib/foremanCentosJob.groovy{empty}
+        - pipelines/lib/pipelines.groovy{empty}
+        - ../theforeman.org/pipelines/lib/duffy.groovy{empty}
+        - ../theforeman.org/pipelines/lib/ansible.groovy{empty}
+        - ../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}
 
 - project:
-    name: katello
+    name: katello-release-3.14-3.15
     jobs:
-      - 'foreman-katello-{version}-test'
-      - 'foreman-katello-upgrade-{version}-test'
+      - 'foreman-katello-{version}-{distro}-{action}-test'
     empty: ''
     version:
-      - '3.13'
       - '3.14'
       - '3.15'
+    distro:
+      - centos7
+    action:
+      - install
+      - upgrade
+
+- project:
+    name: katello-release-3.16
+    jobs:
+      - 'foreman-katello-{version}-{distro}-{action}-test'
+    empty: ''
+    version:
       - '3.16'
+    distro:
+      - centos7
+      - centos8
```

Comment:
I thought you wanted to leave el8 out for now?
But more so: does that combo means it will try a 3.14→3.15→3.16 upgrade on el8 (and fail?)

Reply:
no, it will be just defined, not run, good good